include/renderer_vk/renderer_vk.hpp

```diff
@@ -0,0 +1,57 @@
+#include "renderer.hpp"
+#include "vulkan_api.hpp"
+
+class GPU;
+
+class RendererVK final : public Renderer {
+	SDL_Window* targetWindow;
+
+	// The order of these `Unique*` members is important, they will be destroyed in RAII order
+	vk::UniqueInstance instance = {};
```

Comment:
I will try seeing if I can assert this at compile time via static_assert + offsetof so nobody accidentally breaks destructor order.

Reply:
This may not be necessary to maintain since Vulkan validation layers will already yell at you if you destroy things in the wrong order. Like destroying an instance or device or command-pool while dependent objects still exist and such.